ibis/common/annotations.py

```diff
@@ -22,64 +23,73 @@
 VAR_POSITIONAL = inspect.Parameter.VAR_POSITIONAL
 
 
+class ValidationError(Exception):
```

Comment:
We may want to move this exception to `ibis.common.exceptions` but I'm deferring that to a follow-up.

Reply:
SGTM!